web_m2x_options/static/src/components/form.esm.js

```diff
@@ -0,0 +1,342 @@
+        const optionLimit = Number(options.limit);
+        const irOptionLimit = Number(session.web_m2x_options["web_m2x_options.limit"]);
+
+        if (optionLimit > 1) {
+            props.searchLimit = optionLimit - 1;
+        } else if (irOptionLimit > 1) {
+            props.searchLimit = irOptionLimit - 1;
+        }
+
+        return props;
+    },
+
+    m2mOptionsPropsSearchMore(props, attrs, options) {
+        const irOptionsSearchMore =
+            session.web_m2x_options["web_m2x_options.search_more"];
+
+        if (options.search_more === false || irOptionsSearchMore === "False") {
+            props.noSearchMore = true;
+        } else if (options.search_more || irOptionsSearchMore === "True") {
+            props.noSearchMore = false;
+        }
+
+        return props;
+    },
+
+    m2mOptionsProps(props, attrs, options) {
+        const updatedProps = {
+            ...props,
+            ...this.m2mOptionsPropsCreate(props, attrs, options),
+            ...this.m2mOptionsPropsCreateEdit(props, attrs, options),
+            ...this.m2mOptionsPropsLimit(props, attrs, options),
+            ...this.m2mOptionsPropsSearchMore(props, attrs, options),
+            fieldColor: options.field_color,
+            fieldColorOptions: options.colors,
+        };
+
+        return updatedProps;
+    },
+
+    extractProps({attrs, options, string}, dynamicInfo) {
+        const props = super.extractProps({attrs, options, string}, dynamicInfo);
+        const new_props = this.m2mOptionsProps(props, attrs, options);
+        return new_props;
+    },
+});
+
+patch(Many2XAutocomplete.prototype, {
+    async loadOptionsSource(request) {
+        const options = await super.loadOptionsSource(request);
+        const {fieldColor, fieldColorOptions: colors, resModel} = this.props;
+
+        if (fieldColor && colors) {
+            const valueIds = options.map((result) => result.value);
+
+            const objects = await this.orm.call(resModel, "search_read", [], {
+                domain: [["id", "in", valueIds]],
+                fields: [fieldColor],
+            });
+
+            const objectMap = objects.reduce((map, obj) => {
+                map[obj.id] = obj[fieldColor];
+                return map;
+            }, {});
+
+            options.forEach((option) => {
+                const colorKey = objectMap[option.value];
+                if (colorKey) {
+                    const color = colors[colorKey] || "black";
+                    option.style = `color:${color}`;
+                }
+            });
+        }
+        return options;
+    },
+});
+
+patch(FormController.prototype, {
+    /**
+     * @override
+     */
+    setup() {
+        super.setup();
+        this._setSubViewLimit();
+    },
+    /**
+     * @override
+     * add more method to add subview limit on formview
+     */
+    async _setSubViewLimit() {
+        const ir_options = session.web_m2x_options || {};
```

Comment:
I see you are trying to convert variable and function declaration style to `camelCase`, so why is this variable out of sync with that convention?